Memory Leak investigation

Hi, first off - thank you very much for sharing this! This is super helpful.
I've been putting this model to work on AWS Sagemaker for about a week now on multiple instances - and every few hours i see the memory of an instance is maxed out and the worker is killed. see graph:
<img width="1743" alt="Screenshot 2023-07-31 at 9 57 05" src="https://github.com/guillaumekln/faster-whisper/assets/8166910/2429ad0a-6e64-4e40-baf6-0c4f413a19ae">

This is what i'm seeing in Sagemaker logs for each drop you see in the memory graph(screenshot):

"[ERROR] Worker (pid:25) was sent SIGKILL! Perhaps out of memory?"

I've been able to recreate this on an EC2 machine running CUDA
i am using **ml.g4dn.xlarge** instance type python 3.10 and CUDA 11

This is the code i am using for the transcription:

`self.model = WhisperModel("small", device="cuda")

segments, _ = self.model.transcribe(audio=audio_bytes, language="en", initial_prompt=initial_prompt,
                                            no_speech_threshold=0.2)
        transcript_text = ''.join([segment.text for segment in segments])`
        
*I am initializing the model once upon instance deployment

Would appreciate any assistance with this issue - the performance is amazing!

Thanks in advance

--- a/faster_whisper/audio.py
+++ b/faster_whisper/audio.py
@@ -6,6 +6,7 @@ system dependencies. FFmpeg does not need to be installed on the system.
 However, the API is quite low-level so we need to manipulate audio frames directly.
 """
 
+import gc
 import io
 import itertools
 
@@ -52,6 +53,11 @@ def decode_audio(
             array = frame.to_ndarray()
             dtype = array.dtype
             raw_buffer.write(array)
+
+    # It appears that some objects related to the resampler are not freed
+    # unless the garbage collector is manually run.
+    del resampler
+    gc.collect()
 
     audio = np.frombuffer(raw_buffer.getbuffer(), dtype=dtype)
 

